### PR TITLE
fix(express-oie-sample): id-first auth flows - OKTA-485542

### PR DESCRIPTION
**Pull request by shuowu**
* fixes id first scenarios in express oie sample
* add id-first auth feature in test
* supports loading different issuers based on cucumber `given` step

### CODE REVIEW

File: scripts/e2e-express-embedded-auth-with-sdk.sh [hunk 1]
```diff
@@ -25,19 +25,23 @@ get_secret prod/okta-sdk-vars/fb_password FB_PASSWORD
...
   get_vault_secret_key devex/trex-js-idx-sdk-vars trex_mfa_client_id CUSTOM_CLIENT_ID
   get_vault_secret_key devex/trex-js-idx-sdk-vars trex_mfa_client_secret CUSTOM_CLIENT_SECRET
 else
   echo "Running tests against production (ok12) org"
   export ISSUER=https://javascript-idx-sdk.okta.com
+  export ISSUER_IDFIRST=https://javascript-idx-sdk-idfirst.okta.com
   export CLIENT_ID=0oav2oxnlYjULp0Cy5d6
   get_vault_secret_key devex/prod-js-idx-sdk-vars prod_client_secret CLIENT_SECRET
   get_vault_secret_key devex/prod-js-idx-sdk-vars prod_idx_sdk_org_api_key OKTA_API_KEY
+  get_vault_secret_key devex/prod-js-idx-sdk-vars prod_idx_idfirst_sdk_org_api_key OKTA_API_KEY_IDFIRST
```
Comment (shuowu) [inline scripts/e2e-express-embedded-auth-with-sdk.sh:44]: @vijetmahabaleshwar-okta prod and trex org and vault updates will be needed based on changes here.

**Review by jaredperreault-okta — APPROVED** [review]

File: scripts/e2e-express-embedded-auth-with-sdk.sh (hunk 1 above)
Comment (vijetmahabaleshwar-okta) [reply scripts/e2e-express-embedded-auth-with-sdk.sh:44]: Added the API Token for id-first org in `prod_idx_idfirst_sdk_org_api_key` 
This should pass on bacon now

File: samples/generated/express-embedded-auth-with-sdk/web-server/routes/authenticator.js [hunk 2]
```diff
@@ -122,6 +122,29 @@ router.post('/enroll-authenticator/okta_email', async (req, res, next) => {
   handleTransaction({ req, res, next, authClient, transaction });
 });
 
+// Handle challenge authenticator -- password
+
+router.get('/challenge-authenticator/okta_password', (req, res) => {
+  renderPage({
+    req, res,
+    render: () => renderTemplate(req, res, 'authenticator', {
+      title: 'Challenge password authenticator',
```
Comment (aarongranick-okta) [inline samples/generated/express-embedded-auth-with-sdk/web-server/routes/authenticator.js:131]: Should we use a more user friendly title? "Enter your password"

**Review by aarongranick-okta — APPROVED** [review]

File: samples/config.js [hunk 3]
```diff
@@ -94,12 +94,14 @@ const samples = [
...
       'root-page', 
       'basic-auth', 
+      'identifier-first-auth',
       'self-service-password-recovery', 
       'self-service-registration-custom-attribute',
       'self-service-registration-activation-token',
       'mfa-password-and-email',
       'mfa-password-and-email-magic-link',
-      'social-login-mfa',
+      // This feature is not well defined and introduce flakiness, disable it
```
Comment (shuowu) [inline samples/config.js:103]: Confirmed with @vijetmahabaleshwar-okta

**Pull request closed without merging**